mmaction/datasets/audio_dataset.py

```diff
@@ -0,0 +1,135 @@
+import os.path as osp
+
+import torch
+from mmcv.utils import print_log
+
+from ..core import mean_class_accuracy, top_k_accuracy
+from .base import BaseDataset
+from .registry import DATASETS
+
+
+@DATASETS.register_module()
+class AudioDataset(BaseDataset):
+    """Audio dataset for video recognition. Extracts the audio feature on-the-
+    fly. Annotation file can be that of the rawframe dataset, or:
+
+    .. code-block:: txt
+
+        some/directory-1.wav 163 1
+        some/directory-2.wav 122 1
+        some/directory-3.wav 258 2
+        some/directory-4.wav 234 2
+        some/directory-5.wav 295 3
+        some/directory-6.wav 121 3
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        suffix (str): The suffix of the audio file. Default: '.wav'.
+        kwargs (dict): Other keyword args for `BaseDataset`.
+    """
+
+    def __init__(self, ann_file, pipeline, suffix='.wav', **kwargs):
+        self.suffix = suffix
+        super().__init__(ann_file, pipeline, modality='Audio', **kwargs)
+
+    def load_annotations(self):
+        """Load annotation file to get video information."""
+        if self.ann_file.endswith('.json'):
+            return self.load_json_annotations()
+        video_infos = []
+        with open(self.ann_file, 'r') as fin:
+            for line in fin:
+                line_split = line.strip().split()
+                video_info = {}
+                idx = 0
+                filename = line_split[idx]
+                if self.data_prefix is not None:
+                    if not filename.endswith(self.suffix):
+                        filename = osp.join(self.data_prefix,
+                                            filename) + self.suffix
+                    else:
+                        filename = osp.join(self.data_prefix, filename)
+                video_info['audio_path'] = filename
+                idx += 1
+                # idx for total_frames
+                video_info['total_frames'] = int(line_split[idx])
+                idx += 1
+                # idx for label[s]
+                label = [int(x) for x in line_split[idx:]]
+                assert len(label), f'missing label in line: {line}'
+                if self.multi_class:
+                    assert self.num_classes is not None
+                    onehot = torch.zeros(self.num_classes)
+                    onehot[label] = 1.0
+                    video_info['label'] = onehot
+                else:
+                    assert len(label) == 1
+                    video_info['label'] = label[0]
+                video_infos.append(video_info)
+
+        return video_infos
```

Comment:
“information” is an uncountable noun.

use var name `video_info` for the set and use the name `vid_info` or `info` for a piece of information?

Reply:
> looks good generally. two comments:
> 
> 1. file size of `tests/data/test.wav` is too large, you can trim it to a smaller one
> 2. on dealing with files with no audio:
>    a. how about drop them from the training set? if there are no input, why there should be any output?
>    b. dummy data vs silent:
>    
>    * i think all zero is an valid input when there are no audios.
>    * if you prefer generating random signal, then have you checked that the scale of the signal are similar to a valid audio? if not, how could they be better on preventing overfitting?

2.a. This is because for later usage, I would like the audio and visual share the same dataset, so that I can generate time-syncronized pairs from each videos.
2.b. Yes, there is a mistake here. The audio feature is normalized to 0-1. But the `randn` generates a distribution of mean 0 and 1 std-var(there maybe negative numbers). I will train a new one with the right random signal soon.

For now, FAIR hasn't released the code for AVSlowfast, and I am not sure how they handle and report the accuracy when no audio exists.